datahub-web-react/src/app/tags/CreateNewTagModal/index.tsx

```diff
@@ -0,0 +1,159 @@
+
+    /**
+     * Handler for creating the tag and applying it to entities
+     */
+    const onOk = async () => {
+        if (!canCreateTags) {
+            message.error('You do not have permission to create tags');
+            return;
+        }
+
+        if (!tagName) {
+            message.error('Tag name is required');
+            return;
+        }
+
+        setIsLoading(true);
+
+        try {
+            // Step 1: Create the new tag
+            const createTagResult = await createTagMutation({
+                variables: {
+                    input: {
+                        id: tagName,
+                        name: tagName,
+                        description: tagDescription,
+                    },
+                },
+            });
+
+            const newTagUrn = createTagResult.data?.createTag;
+
+            if (!newTagUrn) {
+                throw new Error('Failed to create tag: No URN returned');
+            }
+
+            // Step 2: Add color
+            if (tagColor) {
+                await setTagColorMutation({
+                    variables: {
+                        urn: newTagUrn,
+                        colorHex: tagColor,
+                    },
+                });
+            }
+
+            // Step 3: Add owners if any
+            if (pendingOwners.length > 0) {
+                await batchAddOwnersMutation({
+                    variables: {
+                        input: {
+                            owners: pendingOwners,
+                            resources: [{ resourceUrn: newTagUrn }],
+                        },
+                    },
+                });
+            }
+
+            message.success(`Tag "${tagName}" successfully created`);
+            onClose();
+        } catch (e: any) {
+            message.destroy();
+            message.error({ content: `Failed to create tag: \n ${e.message || ''}`, duration: 3 });
+        } finally {
+            setIsLoading(false);
+        }
+    };
+
+    // Handle the Enter press
+    useEnterKeyListener({
+        querySelectorToExecuteClick: '#createNewTagButton',
+    });
+
+    // Modal buttons configuration
+    const buttons: ModalButton[] = [
+        {
+            text: 'Cancel',
+            color: 'violet',
+            variant: 'text',
+            onClick: onClose,
+        },
+        {
+            text: 'Create',
+            id: 'createNewTagButton',
+            color: 'violet',
+            variant: 'filled',
+            onClick: onOk,
+            disabled: !tagName || isLoading || !canCreateTags,
+            isLoading,
+        },
+    ];
+
+    // Render permission denied modal if user can't create tags
+    if (!canCreateTags) {
+        return (
+            <Modal title="Create New Tag" onCancel={onClose} buttons={buttons} open={open} centered width={400}>
+                <p>You don&apos;t have permission to create tags. Please contact your DataHub administrator.</p>
```

Comment:
You should do this stuff before the user even gets into the modal ideally, in the parent ManageTags component!